// ngx-summernote/esm2020/lib/ngx-summernote-view.directive.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class NgxSummernoteViewDirective {
    constructor(renderer2, element) {
        this.renderer2 = renderer2;
        this._element = element.nativeElement;
    }
    set ngxSummernoteView(content) {
        this._element.innerHTML = content || '';
    }
    ngAfterViewInit() {
        this.renderer2.addClass(this._element, 'ngx-summernote-view');
    }
}
NgxSummernoteViewDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteViewDirective, deps: [{ token: i0.Renderer2 }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
NgxSummernoteViewDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.3.0", type: NgxSummernoteViewDirective, selector: "[ngxSummernoteView]", inputs: { ngxSummernoteView: "ngxSummernoteView" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteViewDirective, decorators: [{
            type: Directive,
            args: [{
                    // tslint:disable-next-line:directive-selector
                    selector: '[ngxSummernoteView]'
                }]
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: i0.ElementRef }]; }, propDecorators: { ngxSummernoteView: [{
                type: Input
            }] } });
