// ngx-summernote/esm2020/lib/ngx-summernote.module.mjs
import { NgModule } from '@angular/core';
import { NgxSummernoteDirective } from './ngx-summernote.directive';
import { NgxSummernoteViewDirective } from './ngx-summernote-view.directive';
import * as i0 from "@angular/core";
export class NgxSummernoteModule {
    static forRoot() {
        return { ngModule: NgxSummernoteModule, providers: [] };
    }
}
NgxSummernoteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxSummernoteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteModule, declarations: [NgxSummernoteDirective,
        NgxSummernoteViewDirective], exports: [NgxSummernoteDirective,
        NgxSummernoteViewDirective] });
NgxSummernoteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.3.0", ngImport: i0, type: NgxSummernoteModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxSummernoteDirective,
                        NgxSummernoteViewDirective
                    ],
                    exports: [
                        NgxSummernoteDirective,
                        NgxSummernoteViewDirective
                    ]
                }]
        }] });
